addons/event/models/event_mail.py

```diff
@@ -195,12 +195,15 @@ def _execute_attendee_based(self):
         new_attendee_domain = [
             ('event_id', '=', self.event_id.id),
             ("state", "not in", ("cancel", "draft")),
-            ("id", "not in", self.mail_registration_ids.registration_id.ids),
+            ("id", "not in", self.env["event.registration"]._search([
```

Comment:
Did some experimenting this morning on this request.

It generates the following condition currently (checked with XDO profiler):
```sql
event_registration.id NOT IN
(
    SELECT event_registration.id
    FROM event_registration
    WHERE ((event_registration.active = true)
    AND (event_registration.id IN (
        SELECT event_mail_registration.registration_id FROM event_mail_registration 
        WHERE (event_mail_registration.scheduler_id IN (24))))
    )
)
```

If I change the code to:
```py
("mail_registration_ids", "not in", self.env["event.mail.registration"]._search(
    [('scheduler_id', 'in', self.ids)]
)),
```

It becomes:
```sql
event_registration.id NOT IN
(
    SELECT event_mail_registration.registration_id FROM event_mail_registration
    WHERE (event_mail_registration.scheduler_id IN (24))
)
```

Which looks like it does the same thing, will less steps?